pulumi/src/api.ts

```diff
@@ -18,33 +19,51 @@ export interface ApiConfig {
 
 export interface DeployApiArgs {
   appName: string
-  coinstack: string
-  assetName: string
-  baseImageName?: string
-  buildAndPushImageArgs: Pick<BuildAndPushImageArgs, 'context' | 'dockerFile'>
+  assetName: string,
+  coinstack: string,
+  coinstackType: CoinstackType,
+  baseImageName?: string,
+  sampleEnv: Buffer,
   config: Pick<Config, 'api' | 'dockerhub' | 'rootDomainName' | 'environment'>
-  container: Partial<Pick<k8s.types.input.core.v1.Container, 'args' | 'command'>>
   deployDependencies?: Input<Array<Resource>>
-  getHash: (coinstack: string, buildArgs: Record<string, string>) => Promise<string>
   namespace: string
   provider: k8s.Provider
-  secretEnvs: (coinstack: string, asset: string) => k8s.types.input.core.v1.EnvVar[]
+}
+
+const getbuildAndPushImageArgs = (coinstackType: CoinstackType, coinstack: string) => {
+
+  const nodeBasePath = `../../node/${coinstack}`
+  const goBasePath = `../../go/${coinstack}`
+
+  switch(coinstackType){
+    case CoinstackType.GO:  
+      return { context: `${goBasePath}/go`, dockerFile: `${goBasePath}/build/Dockerfile` }
+    case CoinstackType.NODE:
+      return { context: `${nodeBasePath}/api` }
+  }
+}
```

Comment:
The big problem I still see is having these path related code bits all over the place. Still trying to figure out the best way to solve that